### PR TITLE
Add Urgent/11 vulnerability scanner

### PR DESCRIPTION
**Pull request by busterb**
This is a quick port of the Urgent/11 vulnerability scanner from https://github.com/ArmisSecurity/urgent11-detector as an external Metasploit module. This set of vulnerabilities has been getting a lot of attention lately, so having a scanner in Metasploit might be useful.

## Verification

- [ ] Start `msfconsole`
- [ ] `use auxiliary/scanner/discovery/urgent11`
- [ ] Set rhosts / rport to proper target values
- [ ] **Verify** the module can identify a vulnerable target
- [ ] **Verify** the thing does not identify safe targets

Note, I don't have an actual vulnerable target yet, so I'm not sure if this works in reality. Also, the module wants to do some iptables tweaking, which I made conditional on the python3 platform being Linux, since iptables is Linux-only. This module might also be portable to Ruby directly, but I was more interested in being able to track upstream changes to the detector with minimal effort.

```
msf5 auxiliary(scanner/discovery/urgent11) > run

[*] Running for 127.0.0.1...
[*] Running against 127.0.0.1:8080
[*] 	TcpMalformedOptionsDetection  	VxWorks: -100	IPnet: -100
[*] 	TcpDosDetection               	VxWorks: 0	IPnet: 0
[*] 	IcmpCodeDetection             	VxWorks: 0	IPnet: -20
[*] 	IcmpTimestampDetection        	VxWorks: 0	IPnet: 0
[*] IP 127.0.0.1 detected as NOT IPnet
[*] IP 127.0.0.1 detected as NOT VxWorks
[*] Scanned 1 of 1 hosts (100% complete)
[*] Auxiliary module execution completed
```

### CODE REVIEW

**Comment by busterb** [comment]
On second glance, the license of this module might require some clean-room tweaks, since I don't think we have any AGPL code elsewhere. I _think_ external modules avoid any kind of linking problems, but GPL/AGPL is a bit of a pain. The port blocking code feels like something packetfu should be able to do as well in a more portable way.

File: modules/auxiliary/scanner/discovery/urgent11.py [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+    },
+    'notes': {
+        'AKA': [
+            'Urgent/11'
+            ]
+    }
+}
+
+import abc
+import argparse
```
Comment (acammack-r7) [inline modules/auxiliary/scanner/discovery/urgent11.py]: This looks superfluous.

Comment (busterb) [reply modules/auxiliary/scanner/discovery/urgent11.py]: sure, that can go

File: modules/auxiliary/scanner/discovery/urgent11.py [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+    'date': '2019-09-19',
+    'references': [
+        {'type': 'url', 'ref': 'https://armis.com/urgent11'},
+        {'type': 'url', 'ref': 'https://github.com/ArmisSecurity/urgent11-detector'},
+        {'type': 'cve', 'ref': '2019-12258'}
+    ],
+    'type': 'single_scanner',
+    'options':  {
+        'rport': {'type': 'port', 'description': 'The target port', 'required': True, 'default': 80},
+    },
```
[the remark this answers is not in the document]
Comment (busterb) [reply modules/auxiliary/scanner/discovery/urgent11.py]: Why is that? The module already ends up with RHOSTS by virtue of being a single_scanner module type and seems to work without it:

```
Module options (auxiliary/scanner/discovery/urgent11):

   Name     Current Setting  Required  Description
   ----     ---------------  --------  -----------
   RHOSTS   127.0.0.1        yes       The target host(s), range CIDR identifier, or hosts file with syntax 'file:<path>'
   THREADS  1                yes       The number of concurrent threads
   rport    80               yes       The target port
```

Did you load it and RHOSTS didn't show up. Curious if there's something in my environment that's different than yours.

**Comment by busterb** [comment]
Note: this module is getting rewritten to avoid portability & license issues.

**Comment by busterb** [comment]
We now have verification that this works against a real target, woo hoo!

**Comment by bwatters-r7** [comment]
# Release Notes
This PR add a scanner for the recent URGENT/11 vulnerability allowing attackers to exploit certain devices relying on a third-party network packet parser, IPnet.

**Pull request merged**